cornice/tests/test_validation.py

```diff
@@ -240,10 +250,17 @@ def test_content_type_with_callable(self):
         response = app.post('/service6', headers={'Content-Type': 'audio/*'},
                             status=415)
         error_description = response.json['errors'][0]['description']
-        self.assertTrue('text/xml' in error_description)
-        self.assertTrue('application/json' in error_description)
+        self.assertIn('text/xml', error_description)
+        self.assertIn('application/json', error_description)
 
-        app.post('/service6', headers={'Content-Type': 'text/xml'})
+    def test_content_type_with_callable_returning_scalar(self):
+        # test that using a callable for content_type works as well
+        # now, the callable returns a scalar instead of a list
+        app = TestApp(main({}))
+        response = app.put('/service6', headers={'Content-Type': 'audio/*'},
+                           status=415)
+        error_description = response.json['errors'][0]['description']
```

Comment:
after seing this twice, I wonder if we should make it a helper function that we could put in cornice/tests/support.py instead. What do you think ? (this can be done in a later pull request if you want)